presto-spark-base/src/main/java/com/facebook/presto/spark/planner/IterativePlanFragmenter.java

```diff
@@ -265,9 +265,13 @@ public PlanNode visitExchange(ExchangeNode node, RewriteContext<FragmentProperti
         public PlanNode visitRemoteSource(RemoteSourceNode node, RewriteContext<FragmentProperties> context)
         {
             List<SubPlan> childSubPlans = node.getSourceFragmentIds().stream()
-                    .map(id -> subPlanByFragmentId.get(id))
+                    .map(subPlanByFragmentId::get)
                     .collect(toImmutableList());
             context.get().addChildren(childSubPlans);
+
+            // the partitioning handle should be the same as the handle from the partitioning scheme
+            // of any of the input fragments.
+            setDistributionForExchange(node.getExchangeType(), childSubPlans.get(0).getFragment().getPartitioningScheme(), context);
```

Comment:
Does the childSubplans will have more than 1 element ?

Reply:
it can, but we don't care which subplan we use

Reply:
no there can't.  RemoteSourceNode needs to refer to at least one remote source. sorry, i merged already, but would you like me to add some validation to ensure that that's true..

Reply:
This is good, no changes are required :)